dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreCollectionQueryBuilder.cs

```diff
@@ -0,0 +1,195 @@
+            Verify.NotNullOrWhiteSpace(partitionKey);
+
+            queryDefinition.WithParameter($"{RecordKeyVariableName}{i}", recordKey);
+            queryDefinition.WithParameter($"{PartitionKeyVariableName}{i}", partitionKey);
+        }
+
+        return queryDefinition;
+    }
+
+    #region private
+
+    private static AzureCosmosDBNoSQLFilter? BuildSearchFilter(
+        VectorSearchFilter? filter,
+        Dictionary<string, string> storagePropertyNames)
+    {
+        const string EqualOperator = "=";
+        const string ArrayContainsOperator = "ARRAY_CONTAINS";
+        const string ConditionValueVariableName = "@cv";
+
+        var tableVariableName = AzureCosmosDBNoSQLConstants.TableQueryVariableName;
+
+        var filterClauses = filter?.FilterClauses.ToList();
+
+        if (filterClauses is not { Count: > 0 })
+        {
+            return null;
+        }
+
+        var whereClauseArguments = new List<string>();
+        var queryParameters = new Dictionary<string, object>();
+
+        for (var i = 0; i < filterClauses.Count; i++)
+        {
+            var filterClause = filterClauses[i];
+
+            string queryParameterName = $"{ConditionValueVariableName}{i}";
+            object queryParameterValue;
+            string whereClauseArgument;
+
+            if (filterClause is EqualToFilterClause equalToFilterClause)
+            {
+                var propertyName = GetStoragePropertyName(equalToFilterClause.FieldName, storagePropertyNames);
+                whereClauseArgument = $"{tableVariableName}.{propertyName} {EqualOperator} {queryParameterName}";
+                queryParameterValue = equalToFilterClause.Value;
+            }
+            else if (filterClause is AnyTagEqualToFilterClause anyTagEqualToFilterClause)
+            {
+                var propertyName = GetStoragePropertyName(anyTagEqualToFilterClause.FieldName, storagePropertyNames);
+                whereClauseArgument = $"{ArrayContainsOperator}({tableVariableName}.{propertyName}, {queryParameterName})";
+                queryParameterValue = anyTagEqualToFilterClause.Value;
+            }
+            else
+            {
+                throw new NotSupportedException(
+                    $"Unsupported filter clause type '{filterClause.GetType().Name}'. " +
+                    $"Supported filter clause types are: {string.Join(", ", [
+                        nameof(EqualToFilterClause),
+                        nameof(AnyTagEqualToFilterClause)])}");
+            }
+
+            whereClauseArguments.Add(whereClauseArgument);
+            queryParameters.Add(queryParameterName, queryParameterValue);
+        }
+
+        return new AzureCosmosDBNoSQLFilter
+        {
+            WhereClauseArguments = whereClauseArguments,
+            QueryParameters = queryParameters,
+        };
+    }
+
+    private static string GetStoragePropertyName(string propertyName, Dictionary<string, string> storagePropertyNames)
```

Comment:
I think we are using `storageName` instead. Please check with @westey-m as this came up in a discussion yesterday.

Reply:
Discussed with @westey-m , it's going to be renamed all together in scope of separate PR.